src/scripts/main.js

```diff
@@ -1,3 +1,24 @@
 'use strict';
 
 // write code here
+const listOfLi = document.querySelectorAll('li');
+
+for (const li of listOfLi) {
+  const product = li.firstChild;
```

Comment:
Ensure that `li.firstChild` is indeed a text node before accessing `textContent`. If the first child is not a text node, this could lead to unexpected behavior. You might want to check `product.nodeType` to ensure it is a text node (Node.TEXT_NODE) before proceeding.